File: Loja/Loja.UI/DtosTs/ClienteDto/DadosClienteCadastroDto.ts
import { EnderecoEntregaClienteCadastroDto } from "./EnderecoEntregaClienteCadastroDto";

export class DadosClienteCadastroDto {
    Loja: string;
    Indicador_Orcamentista: string;
    Vendedor: string;
    Id: string;
    Cnpj_Cpf: string;
    Rg: string;
    Ie: string;
    Contribuinte_Icms_Status: number;
    Tipo: string;
    Observacao_Filiacao: string;
    Nascimento: Date | string | null;
    Sexo: string;
    Nome: string;
    ProdutorRural: number;
    Endereco: string;
    Numero: string;
    Complemento: string;
    Bairro: string;
    Cidade: string;
    Uf: string;
    Cep: string;
    DddResidencial: string;
    TelefoneResidencial: string;
    DddComercial: string;
    TelComercial: string;
    Ramal: string;
    DddCelular: string;
    Celular: string;
    TelComercial2: string;
    DddComercial2: string;
    Ramal2: string;
    Email: string;
    EmailXml: string;
    Contato: string;

    public static DadosClienteCadastroDtoDeEnderecoEntregaDtoClienteCadastro(end: EnderecoEntregaClienteCadastroDto): DadosClienteCadastroDto {
        let dados: DadosClienteCadastroDto = new DadosClienteCadastroDto();
        dados.Cnpj_Cpf = end.EndEtg_cnpj_cpf;
        dados.Rg = end.EndEtg_rg;
        dados.Ie = end.EndEtg_ie;
        dados.Contribuinte_Icms_Status = end.EndEtg_contribuinte_icms_status;
        dados.Tipo = end.EndEtg_tipo_pessoa;
        dados.Nome = end.EndEtg_nome;
        dados.ProdutorRural = end.EndEtg_produtor_rural_status;
        dados.DddResidencial = end.EndEtg_ddd_res;
        dados.TelefoneResidencial = end.EndEtg_tel_res;
        dados.DddComercial = end.EndEtg_ddd_com;
        dados.TelComercial = end.EndEtg_tel_com;
        dados.Ramal = end.EndEtg_ramal_com != null ? end.EndEtg_ramal_com : "";
        dados.DddCelular = end.EndEtg_ddd_cel;
        dados.Celular = end.EndEtg_tel_cel;
        dados.DddComercial2 = end.EndEtg_ddd_com_2;
        dados.TelComercial2 = end.EndEtg_tel_com_2;
        dados.Ramal2 = end.EndEtg_ramal_com_2 != null ? end.EndEtg_ramal_com_2 : "";
        dados.Email = end.EndEtg_email;
        dados.EmailXml = end.EndEtg_email_xml;

        return dados;
    }
}

File: Global/Prepedido/Dto/DadosClienteCadastroDto.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace Prepedido.Dto
{
    public class DadosClienteCadastroDto
    {
        public string Loja { get; set; }
        public string Indicador_Orcamentista { get; set; }
        public string Vendedor { get; set; }
        public string Id { get; set; }
        public string Cnpj_Cpf { get; set; }
        public string Rg { get; set; }
        public string Ie { get; set; }
        public byte Contribuinte_Icms_Status { get; set; }
        public string Tipo { get; set; }
        public string Observacao_Filiacao { get; set; }
        public string Sexo { get; set; }
        public string Nome { get; set; }
        public byte ProdutorRural { get; set; }
        public string Endereco { get; set; }
        public string Numero { get; set; }
        public string Complemento { get; set; }
        public string Bairro { get; set; }
        public string Cidade { get; set; }
        public string Uf { get; set; }
        public string Cep { get; set; }
        public string DddResidencial { get; set; }
        public string TelefoneResidencial { get; set; }
        public string DddComercial { get; set; }
        public string TelComercial { get; set; }
        public string Ramal { get; set; }
        public string DddCelular { get; set; }
        public string Celular { get; set; }
        public string TelComercial2 { get; set; }
        public string DddComercial2 { get; set; }
        public string Ramal2 { get; set; }
        public string Email { get; set; }
        public string EmailXml { get; set; }
        public string Contato { get; set; }
        public string UsuarioCadastro { get; set; }
        public int? IdOrcamentoCotacao { get; set; }
        public int? IdIndicadorVendedor { get; set; }
        public float? Perc_max_comissao_padrao { get; set; }
        public float? Perc_max_comissao_e_desconto_padrao { get; set; }

        public static DadosClienteCadastroDto DadosClienteCadastroDto_De_DadosClienteCadastroDados(Cliente.Dados.DadosClienteCadastroDados origem)
        {
            if (origem == null) return null;
            return new DadosClienteCadastroDto()
            {
                Loja = origem.Loja,
                Indicador_Orcamentista = origem.Indicador_Orcamentista,
                Vendedor = origem.Vendedor,
                Id = origem.Id,
                Cnpj_Cpf = origem.Cnpj_Cpf,
                Rg = origem.Rg,
                Ie = origem.Ie,
                Contribuinte_Icms_Status = origem.Contribuinte_Icms_Status,
                Tipo = origem.Tipo,
                Observacao_Filiacao = origem.Observacao_Filiacao,
                Sexo = origem.Sexo,
                Nome = origem.Nome,
                ProdutorRural = origem.ProdutorRural,
                Endereco = origem.Endereco,
                Numero = origem.Numero,
                Complemento = origem.Complemento,
                Bairro = origem.Bairro,
                Cidade = origem.Cidade,
                Uf = origem.Uf,
                Cep = origem.Cep,
                DddResidencial = origem.DddResidencial,
                TelefoneResidencial = origem.TelefoneResidencial,
                DddComercial = origem.DddComercial,
                TelComercial = origem.TelComercial,
                Ramal = origem.Ramal,
                DddCelular = origem.DddCelular,
                Celular = origem.Celular,
                TelComercial2 = origem.TelComercial2,
                DddComercial2 = origem.DddComercial2,
                Ramal2 = origem.Ramal2,
                Email = origem.Email,
                EmailXml = origem.EmailXml,
                Contato = origem.Contato,
                UsuarioCadastro = "",
                IdOrcamentoCotacao = origem.IdOrcamentoCotacao,
                IdIndicadorVendedor = origem.IdIndicadorVendedor,
                Perc_max_comissao_padrao = origem.Perc_max_comissao_padrao,
                Perc_max_comissao_e_desconto_padrao = origem.Perc_max_comissao_e_desconto_padrao
            };
        }

        public static Cliente.Dados.DadosClienteCadastroDados DadosClienteCadastroDados_De_DadosClienteCadastroDto(DadosClienteCadastroDto origem)
        {
            if (origem == null) return null;
            return new Cliente.Dados.DadosClienteCadastroDados()
            {
                Loja = origem.Loja,
                Indicador_Orcamentista = origem.Indicador_Orcamentista,
                Vendedor = origem.Vendedor,
                Id = origem.Id,
                Cnpj_Cpf = origem.Cnpj_Cpf,
                Rg = origem.Rg,
                Ie = origem.Ie,
                Contribuinte_Icms_Status = origem.Contribuinte_Icms_Status,
                Tipo = origem.Tipo,
                Observacao_Filiacao = origem.Observacao_Filiacao,
                Sexo = origem.Sexo,
                Nome = origem.Nome,
                ProdutorRural = origem.ProdutorRural,
                Endereco = origem.Endereco,
                Numero = origem.Numero,
                Complemento = origem.Complemento,
                Bairro = origem.Bairro,
                Cidade = origem.Cidade,
                Uf = origem.Uf,
                Cep = origem.Cep,
                DddResidencial = origem.DddResidencial,
                TelefoneResidencial = origem.TelefoneResidencial,
                DddComercial = origem.DddComercial,
                TelComercial = origem.TelComercial,
                Ramal = origem.Ramal,
                DddCelular = origem.DddCelular,
                Celular = origem.Celular,
                TelComercial2 = origem.TelComercial2,
                DddComercial2 = origem.DddComercial2,
                Ramal2 = origem.Ramal2,
                Email = origem.Email,
                EmailXml = origem.EmailXml,
                Contato = origem.Contato,
                UsuarioCadastro = origem.UsuarioCadastro,
                IdOrcamentoCotacao = origem.IdOrcamentoCotacao,
                IdIndicadorVendedor = origem.IdIndicadorVendedor,
                Perc_max_comissao_padrao = origem.Perc_max_comissao_padrao,
                Perc_max_comissao_e_desconto_padrao = origem.Perc_max_comissao_e_desconto_padrao
            };
        }


    }
}